sdk/azidentity/managed_identity_credential_test.go

```diff
@@ -54,80 +55,102 @@ func (m *mockIMDS) Do(req *http.Request) (*http.Response, error) {
 	panic("no more responses")
 }
 
-func TestManagedIdentityCredential_GetTokenInAzureArcLive(t *testing.T) {
-	if len(os.Getenv(arcIMDSEndpoint)) == 0 {
-		t.Skip()
-	}
-	msiCred, err := NewManagedIdentityCredential(nil)
+func TestManagedIdentityCredential_AzureArc(t *testing.T) {
+	file, err := os.Create(filepath.Join(t.TempDir(), "arc.key"))
```

Comment:
I think you're missing a `defer file.Close()`

Reply:
Thanks, I'll add that. Not sure it will solve the failure on Windows. Apparently `testing` cleanup of temp dirs is flaky: https://go-review.googlesource.com/c/go/+/371296. That's fixed for 1.18 so I'll just skip this test on Windows for now.

Reply:
Might be worth it to see if closing the file fixes this first.  If not then yeah just skip on Windows.